### ISSUE TITLE
Line feeds break parsing of local methods

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#12280]
Line feeds between a local method's return type and its name prevent it from being parsed as a valid local method.

Due to this issue code could be easily broken by formatting.

Test code included at the bottom of this report.

![local functions parsing](https://cloud.githubusercontent.com/assets/5566206/16453603/0f4a5754-3e16-11e6-9d8d-d55c4acdcff1.png)

```cs
class Bug
{
    static void Main()
    {
        int
            foo() => 5;

        int bar() => 5;
    }
}
````
